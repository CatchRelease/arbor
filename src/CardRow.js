import { css } from '@emotion/core';
import styled from '@emotion/styled';
import Box from './Box';

const spaceStyles = ({ theme }) => css`
  padding: ${theme.space.smaller} ${theme.space.regular};
  margin: ${theme.space.smaller} 0;
`;

const CardRow = styled(Box.withComponent('div'))`
  ${spaceStyles};
`;

export default CardRow;
