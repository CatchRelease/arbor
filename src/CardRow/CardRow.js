import Flex from '../Flex';

const CardRow = (props) => <Flex {...props} />;

CardRow.defaultProps = {
  px: 'regular',
  py: 'smallest',
  my: 'smaller',
  alignItems: 'center',
  justifyContent: 'space-between'
};

export default CardRow;
