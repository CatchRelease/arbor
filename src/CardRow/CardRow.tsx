import Flex, { FlexProps } from '../Flex';

const CardRow = (props: FlexProps) => <Flex {...props} />;

CardRow.defaultProps = {
  px: 'regular',
  py: 'smallest',
  my: 'smaller',
  alignItems: 'center',
  justifyContent: 'space-between'
};

export default CardRow;
