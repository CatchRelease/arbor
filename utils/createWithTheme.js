import React from 'react';
import { CacheProvider } from '@emotion/core';
import createCache from '@emotion/cache';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'emotion-theming';
import theme from '../src/theme';

const myCache = createCache();
myCache.compat = true;

export default (Component) =>
  create(
    <CacheProvider value={myCache}>
      <ThemeProvider theme={theme}>{Component}</ThemeProvider>
    </CacheProvider>
  );
