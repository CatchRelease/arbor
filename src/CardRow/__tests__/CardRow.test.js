import React from 'react';
import createWithTheme from '../../../tests/utils/createWithTheme';

import CardRow from '../CardRow';

describe('<CardRow />', () => {
  it('renders a CardRow properly', () => {
    const tree = createWithTheme(<CardRow />);

    expect(tree).toMatchSnapshot();
  });
});
