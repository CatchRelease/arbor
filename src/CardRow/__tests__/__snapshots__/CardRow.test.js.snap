// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardRow /> renders a CardRow properly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
  className="emotion-0 emotion-1"
/>
`;
